src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeFieldInfo.cs

```diff
@@ -46,6 +47,13 @@ internal RuntimeType GetDeclaringTypeInternal()
 
         public override Module Module => GetRuntimeModule();
         public override bool IsCollectible => m_declaringType.IsCollectible;
+
+        public override bool Equals(object? obj) =>
+            obj == (object)this ||
```

Comment:
Same as change to `ReferenceEquals` as https://github.com/dotnet/runtime/pull/73009/files#r935802561